desktop/ui/src/main/java/org/datacleaner/widgets/properties/MultipleInputColumnsPropertyWidget.java

```diff
@@ -362,10 +364,20 @@ public void onRemove(TransformerComponentBuilder<?> transformerJobBuilder) {
     }
 
     @Override
+    protected void onPanelAdd() {
+        super.onPanelAdd();
+        final AnalysisJobBuilder analysisJobBuilder = getAnalysisJobBuilder();
+        analysisJobBuilder.addSourceColumnChangeListener(this);
+        analysisJobBuilder.addTransformerChangeListener(this);
```

Comment:
This was moved from constructor to here - to make sure the add and remove timing is completely orthogonal.